### ISSUE TITLE
0.17.2/0.18.0 failed to compile in macOS Sierra 10.12.4/clang-802.0.41

### ISSUE THREAD

**Issue by number5** [JGRennison/OpenTTD-patches#17]
It works fine for 0.17.0

Xcode version: 8.3.1
Clang version:
```
Apple LLVM version 8.1.0 (clang-802.0.41)
Target: x86_64-apple-darwin16.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Let me know if you need more info.

## How to reproduce it

`./configure`
`make`

```
[SRC] Compiling 3rdparty/squirrel/squirrel/sqapi.cpp
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:627:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/utility:371:70: error:
      expected an identifier or template-id after '::'
                   typename __make_tuple_indices<sizeof...(_Args1)>::type(),
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:19: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                  ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:627:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/utility:372:71: error:
      expected an identifier or template-id after '::'
                   typename __make_tuple_indices<sizeof...(_Args2) >::type())
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:19: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                  ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:605:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:147:85: error:
      expected expression
  ...static _LIBCPP_CONSTEXPR type min() _NOEXCEPT {return type();}
                                                           ^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:24: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                       ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:605:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:147:61: error:
      no return statement in constexpr function
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type min() _NOEXC...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:148:85: error:
      expected expression
  ...static _LIBCPP_CONSTEXPR type max() _NOEXCEPT {return type();}
                                                           ^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:24: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                       ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:605:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:148:61: error:
      no return statement in constexpr function
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type max() _NOEXC...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:149:88: error:
      expected expression
  ...static _LIBCPP_CONSTEXPR type lowest() _NOEXCEPT {return type();}
                                                              ^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:24: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                       ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:605:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:149:61: error:
      no return statement in constexpr function
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type lowest() _NO...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:158:89: error:
      expected expression
  ...static _LIBCPP_CONSTEXPR type epsilon() _NOEXCEPT {return type();}
                                                               ^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:24: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                       ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:605:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:158:61: error:
      no return statement in constexpr function
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type epsilon() _N...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:159:93: error:
      expected expression
  ...static _LIBCPP_CONSTEXPR type round_error() _NOEXCEPT {return type();}
                                                                   ^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:24: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                       ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:605:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:159:61: error:
      no return statement in constexpr function
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type round_error(...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:171:90: error:
      expected expression
  ...static _LIBCPP_CONSTEXPR type infinity() _NOEXCEPT {return type();}
                                                                ^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:24: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                       ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:605:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:171:61: error:
      no return statement in constexpr function
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type infinity() _...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:172:91: error:
      expected expression
  ...static _LIBCPP_CONSTEXPR type quiet_NaN() _NOEXCEPT {return type();}
                                                                 ^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:24: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                       ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:605:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:172:61: error:
      no return statement in constexpr function
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type quiet_NaN() ...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:173:95: error:
      expected expression
  ...static _LIBCPP_CONSTEXPR type signaling_NaN() _NOEXCEPT {return type();}
                                                                     ^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:24: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                       ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqapi.cpp:20:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/../../../string_func.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:439:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:605:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:173:61: error:
      no return statement in constexpr function
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type signaling_Na...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/limits:174:92: error:
      expected expression
  ...static _LIBCPP_CONSTEXPR type denorm_min() _NOEXCEPT {return type();}
                                                                  ^
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/squirrel/squirrel/sqobject.h:99:24: note:
      expanded from macro 'type'
#define type(obj) ((obj)._type)
                       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[1]: *** [3rdparty/squirrel/squirrel/sqapi.o] Error 1
make: *** [all] Error 1
```

**Comment by JGRennison** [comment]
Thanks for reporting this. I've updated the jgrpp branch with a possible fix in cb9ffe1b, can you let me know whether this resolves it?

**Comment by number5** [comment]
@JGRennison looks like it fixed sqapi.cpp 👍 
Now it move on to `aircraft_cmd.cpp`

```
[SRC] Compiling aircraft_cmd.cpp
In file included from /Users/bruce/src/git/OpenTTD-patches/src/aircraft_cmd.cpp:34:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/station_base.h:22:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_map.h:31:
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree.h:1127:8: error:
      expected member name or ';' after declaration specifiers
  void verify() const;
  ~~~~ ^
/usr/include/AssertMacros.h:1339:29: note: expanded from macro 'verify'
                #define verify(assertion) __Verify(assertion)
                                          ^
/usr/include/AssertMacros.h:420:5: note: expanded from macro '__Verify'
                  d...
                  ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/aircraft_cmd.cpp:34:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/station_base.h:22:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_map.h:31:
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree.h:1965:16: error:
      expected unqualified-id
void btree<P>::verify() const {
               ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/aircraft_cmd.cpp:34:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/station_base.h:22:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_map.h:32:
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:105:8: error:
      expected member name or ';' after declaration specifiers
  void verify() const {
  ~~~~ ^
/usr/include/AssertMacros.h:1339:29: note: expanded from macro 'verify'
                #define verify(assertion) __Verify(assertion)
                                          ^
/usr/include/AssertMacros.h:420:5: note: expanded from macro '__Verify'
                  d...
                  ^
In file included from /Users/bruce/src/git/OpenTTD-patches/src/aircraft_cmd.cpp:34:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/station_base.h:22:
In file included from /Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_map.h:32:
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:51:9: error:
      member initializer 'tree_' does not name a non-static data member or base
      class
      : tree_(comp, alloc) {
        ^~~~~~~~~~~~~~~~~~
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:56:9: error:
      member initializer 'tree_' does not name a non-static data member or base
      class
      : tree_(x.tree_) {
        ^~~~~~~~~~~~~~
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:60:29: error:
      use of undeclared identifier 'tree_'; did you mean 'true'?
  iterator begin() { return tree_.begin(); }
                            ^~~~~
                            true
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:60:34: error:
      expected ';' after return statement
  iterator begin() { return tree_.begin(); }
                                 ^
                                 ;
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:61:41: error:
      use of undeclared identifier 'tree_'; did you mean 'true'?
  const_iterator begin() const { return tree_.begin(); }
                                        ^~~~~
                                        true
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:61:46: error:
      expected ';' after return statement
  const_iterator begin() const { return tree_.begin(); }
                                             ^
                                             ;
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:62:27: error:
      use of undeclared identifier 'tree_'; did you mean 'true'?
  iterator end() { return tree_.end(); }
                          ^~~~~
                          true
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:62:32: error:
      expected ';' after return statement
  iterator end() { return tree_.end(); }
                               ^
                               ;
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:63:39: error:
      use of undeclared identifier 'tree_'; did you mean 'true'?
  const_iterator end() const { return tree_.end(); }
                                      ^~~~~
                                      true
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:63:44: error:
      expected ';' after return statement
  const_iterator end() const { return tree_.end(); }
                                           ^
                                           ;
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:64:38: error:
      use of undeclared identifier 'tree_'; did you mean 'true'?
  reverse_iterator rbegin() { return tree_.rbegin(); }
                                     ^~~~~
                                     true
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:64:43: error:
      expected ';' after return statement
  reverse_iterator rbegin() { return tree_.rbegin(); }
                                          ^
                                          ;
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:65:50: error:
      use of undeclared identifier 'tree_'; did you mean 'true'?
  const_reverse_iterator rbegin() const { return tree_.rbegin(); }
                                                 ^~~~~
                                                 true
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:65:55: error:
      expected ';' after return statement
  const_reverse_iterator rbegin() const { return tree_.rbegin(); }
                                                      ^
                                                      ;
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:66:36: error:
      use of undeclared identifier 'tree_'; did you mean 'true'?
  reverse_iterator rend() { return tree_.rend(); }
                                   ^~~~~
                                   true
/Users/bruce/src/git/OpenTTD-patches/src/3rdparty/cpp-btree/btree_container.h:66:41: error:
      expected ';' after return statement
  reverse_iterator rend() { return tree_.rend(); }
                                        ^
                                        ;
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[1]: *** [aircraft_cmd.o] Error 1
make: *** [all] Error 1```

**Comment by JGRennison** [comment]
I've added another fix in 612b0adf, can you let me know how that goes?

**Comment by number5** [comment]
Compiling passed and I started it and load a game, all looks good.

Thanks for all the efforts btw.
